src/dataExplorer/context/resultsView.tsx

```diff
@@ -109,20 +158,204 @@ export const ResultsViewProvider: FC = ({children}) => {
+        setView({
+          ...view,
+          properties: {
+            ...view.properties,
+            xColumn: dataColumn,
+          },
+        })
+        break
+      case 'gauge':
+      case 'single-stat':
+        // These graphs are about calculating a single value.
+        // The graph unions over all `results.parsed.fluxGroupKeyUnion = []`, to try creating a single value.
+        // For the unpivoted data --> it has all field values under a single column `_value`. Then fluxGroupKeyUnion = tagKeys.
+        // For the pivoted data --> it has the data broken into different column per field. Cannot support.
+        if (!hasSingleValueColumn) {
+          dispatch(notify(trySmoothingData(view.properties.type)))
+        }
+        break
+      case 'line-plus-single-stat':
+        // Has the same issue as 'single-stat'. Read above.
+        if (!hasSingleValueColumn) {
+          dispatch(notify(trySmoothingData('single-stat')))
+        }
+        // But the line graph will still work.
+        setView({
+          ...view,
+          properties: {
+            ...view.properties,
+            yColumn: dataColumn,
+            xColumn: timeColumn,
+          },
+        })
+        break
+      case 'check':
+      case 'mosaic':
+      // TODO: Is not working at all.
+      // Unclear if there is a data query results format, in which it will work.
+      case 'geo':
+      // user chooses own column with the appropriate json geo data struct
+      case 'simple-table':
+      case 'table':
+      default:
+        // no change. Keep raw.
+        break
+    }
+  }
 
-    // reset groupby
-    const excludeFromGroupby = NOT_PERMITTED_GROUPBY
-    const groupby = Object.keys(
+  // from data table headers
+  const getSelectorColumns = () => {
+    const excludeFromColumnSelectors = NOT_PERMITTED_NAMES_COLUMN_SELECTOR
+
+    return Object.keys(resultFromParent?.parsed?.table?.columns || {}).filter(
+      columnName => !excludeFromColumnSelectors.includes(columnName)
+    )
+  }
+
+  // from data table headers
+  const getNumericSelectorColumns = () => {
+    const excludeFromColumnSelectors = NOT_PERMITTED_NAMES_COLUMN_SELECTOR
+
+    return Object.keys(resultFromParent?.parsed?.table?.columns || {}).filter(
+      columnName => {
+        const {fluxDataType} = (resultFromParent?.parsed?.table?.columns || {})[
+          columnName
+        ]
+        return (
+          !excludeFromColumnSelectors.includes(columnName) &&
+          PERMITTED_NUMERIC_TYPES_COLUMN.includes(fluxDataType)
+        )
+      }
+    )
+  }
+
+  // from data table headers
+  const defineTimeColumn = () => {
+    const dateTimeColumns = Object.keys(
       resultFromParent?.parsed?.table?.columns || {}
-    ).filter(columnName => !excludeFromGroupby.includes(columnName))
-    setViewOptionsAll({groupby})
+    ).filter(columnName => {
+      const {fluxDataType} = (resultFromParent?.parsed?.table?.columns || {})[
```

Comment:
the expression

```ts
resultFromParent?.parsed?.table?.columns || {}
```
is repeated a lot throughout this code. It might be worth storing that expression into a named variable

Reply:
This destruct pattern in used in methods which are triggered by the `useEffect(() => [resultFromParent])`. Therefore it adds complexity with another declared outer scope variable which additionally needs to be reference-tracked for changes.

So we can do this...but future devs are likely to stumble on it.
(Another reason that TEST ARE INCOMING SHORTLY. 😆 )